docker/tarfile/src.go

```diff
@@ -335,7 +335,7 @@ func (s *Source) prepareLayerData(tarManifest *ManifestItem, parsedConfig *manif
 		if err != nil {
 			return nil, err
 		}
-		if li, ok := unknownLayerSizes[h.Name]; ok {
+		if li, ok := unknownLayerSizes[path.Clean(h.Name)]; ok {
```

Comment:
The same normalization must happen when adding entries to `unknownLayerSizes` a few lines above as well.

Also, the `path.Clean(h.Name)` can be only computed once.

Reply:
done, and done